### ISSUE TITLE
After the last update (2021-08-01) breadcrumbs stopped working

### ISSUE THREAD

**Issue by gfidias** [SkepticMystic/breadcrumbs#74]
VM276:6350 Uncaught (in promise) TypeError: Cannot read property 'path' of null
    at eval (eval at <anonymous> (app.js:1), <anonymous>:6350:49)
    at Array.map (<anonymous>)
    at getFieldValues (eval at <anonymous> (app.js:1), <anonymous>:6350:30)
    at eval (eval at <anonymous> (app.js:1), <anonymous>:6374:39)
    at Array.map (<anonymous>)
    at eval (eval at <anonymous> (app.js:1), <anonymous>:6374:18)
    at Array.map (<anonymous>)
    at getNeighbourObjArr (eval at <anonymous> (app.js:1), <anonymous>:6371:48)
    at async BreadcrumbsPlugin.initGraphs (eval at <anonymous> (app.js:1), <anonymous>:26877:30)
    at async eval (eval at <anonymous> (app.js:1), <anonymous>:26804:35)

**Comment by SkepticMystic** [comment]
Thank you for posting this. Can you message me on Discord, it'll make this easier to debug.
My handle is @SkepticMystic

**Comment by superupon** [comment]
Is this issue still working on? I updated the plugin to 0.10.1, still not work.

**Comment by SkepticMystic** [comment]
Hey @superupon will you message me on Discord. Debugging these issues is usually pretty unqie to each person, so it'll be easiest over there

**Comment by superupon** [comment]
Hi, @SkepticMystic, how I could get connected to you? I have not used Discord before.

**Comment by SkepticMystic** [comment]
That's ok, we can debug it over github then :)
Can you do the following:
1) Send the `data.json` file for breadcrumbs. You can find it in `Vault/.obsidian/plugins/breadcrumbs/`
2) Check the Obsidian console to see if you get any errors when running Breadcrumbs
3) Enable Super Debug mode in the settings of Breadcrumbs, then change to a different note in your vault, and send the console logs that appear in the console

**Comment by superupon** [comment]
[data.txt](https://github.com/SkepticMystic/breadcrumbs/files/6931677/data.txt)
Hi, I have to rename it to "data.txt", since github does not allow me to upload ".json" files. 

For 2, Obsidian console only showed output like this:

loading FileExplorerNoteCount
main.ts:76 loading plugin plantuml
VM122:47731 loading breadcrumbs plugin
DevTools failed to load SourceMap: Could not parse content for app://obsidian.md/pickr.min.js.map: Unexpected end of JSON input

For 3,
VM122:6243 {neighbourObjArr: Array(276)}
VM122:6243 graphs inited
VM122:6243 {pathsArr: Array(1)}
VM122:6243 []
VM122:6243 {sortedTrails: Array(0)}
VM122:6243 {realParents: Array(0), impliedParents: Array(0), realSiblings: Array(0), impliedSiblingsArr: Array(0), realChildren: Array(0), …}

**Comment by SkepticMystic** [comment]
Thanks for sending this.
1) Can you change the Index Note setting to be blank and see if that works. Just leave the text box empty.
2) Do you have the dataview plugin enabled? You don't needi it, but it will narrow down the problem.
2) I think you enabled `Debug Mode`, but can you do it with `Super Debug Mode` instead

**Comment by superupon** [comment]
For 1, I tried, still no breadcrumb showed up.

And for easy to debug, I created a new vault with breadcrumb installed only. 
I created two .md file "test" and "A", A has metadata like this:
---
parent: [[test]]
---

And I opened Super Debug Mode. When I choose test.md, it will output
GetObsMetadataCache: test
VM108:6248 {obs: undefined}
VM108:6248 GetObsMetadataCache: A
VM108:6248 {obs: {…}}
VM108:6248 parent of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 sibling of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 child of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 parent of: A.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 sibling of: A.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 child of: A.md
VM108:6248 {rawValues: Array(1)}

And if I choose A.md
it will output:
GetObsMetadataCache: test
VM108:6248 {obs: undefined}
VM108:6248 GetObsMetadataCache: A
VM108:6248 {obs: {…}}
VM108:6248 parent of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 sibling of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 child of: test.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 parent of: A.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 sibling of: A.md
VM108:6248 {rawValues: Array(1)}
VM108:6248 child of: A.md
VM108:6248 {rawValues: Array(1)}

**Comment by SkepticMystic** [comment]
Can you open up each of those objects and send the content inside of them

**Comment by superupon** [comment]
Output:
```
GetObsMetadataCache: test
VM108:6248 
{obs: undefined}
obs: undefined
__proto__: Object
VM108:6248 GetObsMetadataCache: A
VM108:6248 
{obs: {…}}
obs: {parent: Array(1), position: {…}}
__proto__: Object
VM108:6248 parent of: test.md
VM108:6248 
{rawValues: Array(1)}
rawValues: [undefined]
__proto__: Object
VM108:6248 sibling of: test.md
VM108:6248 
{rawValues: Array(1)}
rawValues: [undefined]
__proto__: Object
VM108:6248 child of: test.md
VM108:6248 
{rawValues: Array(1)}
rawValues: [undefined]
__proto__: Object
VM108:6248 parent of: A.md
VM108:6248 
{rawValues: Array(1)}
rawValues: ["test"]
__proto__: Object
VM108:6248 sibling of: A.md
VM108:6248 
{rawValues: Array(1)}
rawValues: [undefined]
__proto__: Object
VM108:6248 child of: A.md
VM108:6248 
{rawValues: Array(1)}
rawValues: [undefined]
__proto__: Object
```

Should I open those objects more deeper?

**Comment by superupon** [comment]
I tried to open those objects more deeper.
![objs1](https://user-images.githubusercontent.com/7564035/128201176-983ad63b-4439-4b55-aa24-048e0015f80d.png)
![objs2](https://user-images.githubusercontent.com/7564035/128201208-1e48c88f-cbda-4477-a57c-aa2c2abbc849.png)

**Comment by SkepticMystic** [comment]
Thank you, this gives me a better idea of where the problem lies.
I'll try fix it today

**Comment by SkepticMystic** [comment]
@superupon can you check if this is working in 0.10.5, I think I found the bug

**Comment by superupon** [comment]
Great, I have verified in my obsidian. This issue is solved in version 0.10.5!
Thanks for your patience and support. Breadcrumb is heavily used in my daily work. That's a big help :)

**Comment by SkepticMystic** [comment]
Awesome! Glad to hear the issue is resolved, and also to hear that you find value from the plugin 😄 
I'm gonna close out this issue, hopefully any other bugs from the refactor are resolved
